src/BaseBuilder.js

```diff
@@ -130,28 +134,60 @@ export default class BaseBuilder {
    *
    * @param {callback} callback - Function which takes an instance of
    * `LazyTemplateInterface` and returns a template provider class
-   * @return {Builder}
+   * @return {BaseBuilder}
    */
   setLazyTemplateFactory(callback) {
     return new this.constructor(this._state.set('_templateProviderCallback', callback));
   }
 
   /**
-   * Set the error message function in the options object for this type.
-   * Override any existing errors.
+   * Set the error key in the options object. Used to set errors from the API
+   * on a tcomb field. By default, also sets hasError to true if `error` is
+   * truthy and false otherwise. You can disable this behavior by passing in
+   * the `autoHasError` option in the config object.
```

Comment:
it stands for automatically has error?

Reply:
`setHasError` might be even more confusing
`autoSetHasError`?

Reply:
Other ideas:

`defaultHasError` (true by default)
`autoSetHasError` (true by default)
`setHasErrorByDefault` (true by default)
`overrideHasError` (false by default)
`customHasError` (false by default)
`manualHasError` (false by default)

Reply:
Spoke offline: Going to use `overrideHasError` and invert the logic.